### ISSUE TITLE
Cleanup routine throws ArgumentException

### ISSUE THREAD

**Issue by naspert** [oleg-shilo/cs-script#33]
Calling Process.GetProcessById throws if the process with the given id is not running.
cf.
https://github.com/oleg-shilo/cs-script/blob/master/Source/Utils.cs#L288

wrapping the call inside a try/catch block should do the trick...

**Comment by naspert** [comment]
(cf. official doc https://msdn.microsoft.com/en-us/library/76fkb36k(v=vs.110).aspx )

**Comment by oleg-shilo** [comment]
Not sure I follow you. It is already wrapped. I  the very same source code lik you provided::
```C#
try
{
    if (Process.GetProcessById(pid) != null)
        continue; //still running
}
catch
{
    //GetProcessById will throw if pid is not running and Utils.FileDelete handles all unexpected
}
```

**Comment by naspert** [comment]
true I missed that. However this behavior has changed wrt previous versions, increasing the noise level when debugging. If I set the environment var "CSScript_Suspend_Housekeeping" I should avoid entirely this routine, right ? (currently tracing an issue at when my application exits and CSScript exceptions get in the way...)

**Comment by naspert** [comment]
(replying to myself) with the env. variable properly set, I managed to trace the real problem. I will close the issue (but still a non-throwing version would be a nice feature ;)

**Comment by oleg-shilo** [comment]
Well... I agree. I also found it extremely annoying when I have to step over handled exceptions while hunting for a real ones. 
I didn't want to replace it with the all process lookup by id unless there is a good reason for this. But since you raised this issue I will do the replacement.

**Comment by naspert** [comment]
I guess you can replace this call by something like this (which should not throw)
`var tmp = Process.GetProcesses().Select(p => p.Id);`
...
and            
`if (tmp.Contains(pid))
    continue;`

**Comment by oleg-shilo** [comment]
Of course. It is exactly what I am going to do. Another good candidate is this: 
```C#
if (Process.GetProcesses().Any(p => p.Id == pid))
    continue; //still running
```
Another reason for not doing it earlier is that .NET 1.1 didn't have LINQ and `foreach...` really hurts my eyes. :) Yes, yes I know, `foreach` is the fastest option but it....  just kills the fun. :)

But now support for 1.1 is over so many things are becoming much simpler.

**Comment by oleg-shilo** [comment]
Fixed in 58ce13fd9ad6c078a097bea5c0cb01e362bb0e69
Planned for v3.21 release

**Comment by oleg-shilo** [comment]
Fixed in Release v3.21.1
